widgets/src/main/java/org/pentaho/mantle/client/dialogs/folderchooser/FolderTree.java

```diff
@@ -711,4 +705,13 @@ public boolean isCreateRootNode() {
     return values;
   }
 
+  public static String getDefaultSaveLocation() {
+    String homeFolder = getWindowHomeFolder();
+    return ( homeFolder != null ) ? homeFolder : System.getProperty( "user.home", "" );
+  }
+
+  public static native String getWindowHomeFolder()
+    /*-{
+        return window.top.HOME_FOLDER;
```

Comment:
Seems like this will cause a cross-domain error when embedded in another app. I don't think we can use window.top.xxx without a try/catch around it.

Reply:
Ok, I will follow a different approach, instead of getting the home folder using the window I will just request that information using "api/session/userWorkspaceDir".
It should be more feasible.